src/main/java/com/comphenix/protocol/wrappers/WrappedLevelChunkData.java

```diff
@@ -0,0 +1,482 @@
+
+            PACKED_XZ_ACCESSOR = Accessors.getFieldAccessor(posFields.get(0));
+            Y_ACCESSOR = Accessors.getFieldAccessor(posFields.get(1));
+            TYPE_ACCESSOR = Accessors.getFieldAccessor(FuzzyReflection.fromClass(HANDLE_TYPE, true)
+                    .getField(FuzzyFieldContract.newBuilder().typeExact(MinecraftReflection.getBlockEntityTypeClass()).build()));
+            TAG_ACCESSOR = Accessors.getFieldAccessor(FuzzyReflection.fromClass(HANDLE_TYPE, true)
+                    .getField(FuzzyFieldContract.newBuilder().typeExact(MinecraftReflection.getNBTCompoundClass()).build()));
+        }
+
+        public BlockEntityInfo(Object handle) {
+            super(HANDLE_TYPE);
+
+            setHandle(handle);
+        }
+
+        /**
+         * The section-relative X-coordinate of the block entity.
+         *
+         * @return the unpacked X-coordinate.
+         */
+        public int getSectionX() {
+            return (int) PACKED_XZ_ACCESSOR.get(handle) >> 4;
+        }
+
+        /**
+         * Sets the section-relative X-coordinate of the block entity
+         *
+         * @param sectionX the section-relative x coordinate
+         */
+        public void setSectionX(int sectionX) {
+            PACKED_XZ_ACCESSOR.set(handle, sectionX << 4 | getSectionZ());
+        }
+
+        /**
+         * The section-relative Z-coordinate of the block entity.
+         *
+         * @return the unpacked Z-coordinate.
+         */
+        public int getSectionZ() {
+            return (int) PACKED_XZ_ACCESSOR.get(handle) & 0xF;
+        }
+
+        /**
+         * Sets the section-relative Z-coordinate of the block entity
+         *
+         * @param sectionZ the section-relative z coordinate
+         */
+        public void setSectionZ(int sectionZ) {
+            PACKED_XZ_ACCESSOR.set(handle, getSectionX() << 4 | sectionZ);
+        }
+
+        /**
+         * The Y-coordinate of the block entity.
+         *
+         * @return the Y-coordinate.
+         */
+        public int getY() {
+            return (int) Y_ACCESSOR.get(handle);
+        }
+
+        /**
+         * Sets the Y-coordinate of the block entity.
+         *
+         * @param y the new y coordinate
+         */
+        public void setY(int y) {
+            Y_ACCESSOR.set(handle, y);
+        }
+
+        /**
+         * The registry key of the block entity type.
+         *
+         * @return the registry key.
+         */
+        public MinecraftKey getTypeKey() {
+            return WrappedRegistry.getBlockEntityTypeRegistry().getKey(TYPE_ACCESSOR.get(handle));
+        }
+
+        /**
+         * Sets the registry key of the block entity type
+         *
+         * @param typeKey the new block entity type key
+         */
+        public void setTypeKey(MinecraftKey typeKey) {
+            TYPE_ACCESSOR.set(handle, WrappedRegistry.getBlockEntityTypeRegistry().get(typeKey));
```

Comment:
I think moving the registry field into a static final variable wouldn't hurt + remove the getter for the registry and just use WrappedRegistry.get (that way we don't have accessors for some registries and most of them missing)